packages/react/src/__tests__/ReactElement-test.js

```diff
@@ -358,6 +358,28 @@ describe('ReactElement', () => {
     expect(instance.props.fruit).toBe('persimmon');
   });
 
+  // NOTE: We're explicitly not using JSX here. This is intended to test
+  // classic JS without JSX.
+  it('should display a warning when mapping key to a default prop', () => {
+    class Component extends React.Component {
+      render() {
+        return React.createElement('div', null, this.props.key);
```

Comment:
Since you render it, you probably want to assert it's in the DOM in your test as well?

Reply:
> Since you render it, you probably want to assert it's in the DOM in your test as well?

Hmm, I don't think rendering the key here adds anything to the test, so I could just not render it.